src/DynamicSizeList.js

```diff
@@ -271,97 +253,99 @@ const DynamicSizeList = createListComponent({
         return;
       }
 
-      delta += newSize - oldSize;
       itemSizeMap[key] = newSize;
 
       if (!instance.state.scrolledToInitIndex) {
         generateOffsetMeasurements(props, index, instanceProps);
-        instance.forceUpdate();
         return;
       }
+
       const element = ((instance._outerRef: any): HTMLDivElement);
 
       if (
-        instance.state.scrollOffset + instance.props.height >=
-        element.scrollHeight - 10
+        instance.props.height + element.scrollTop >=
+        instanceProps.totalMeasuredSize - 10
       ) {
         generateOffsetMeasurements(props, index, instanceProps);
-        instance.forceUpdate();
         instance.scrollToItem(0, 'end');
-        return;
-      }
-
-      const [, , , visibleStopIndex] = instance._getRangeToRender(
-        element.scrollTop
-      );
-
-      generateOffsetMeasurements(props, index, instanceProps);
-
-      if (index < visibleStopIndex - 1) {
         instance.forceUpdate();
         return;
       }
 
-      instance._scrollCorrectionInProgress = true;
+      if (forceScrollCorrection) {
+        const delta = newSize - oldSize;
+        const [, , visibleStartIndex] = instance._getRangeToRender(
+          element.scrollTop
+        );
+        generateOffsetMeasurements(props, index, instanceProps);
+        if (index < visibleStartIndex + 1) {
```

Comment:
is it possible for visibleStartIndex to be undefined? just to be sure, lets handle that case.

Reply:
Nope we will not have undefined for visibleStartIndex it will be `0` at the bare minimum